src/sentry/api/event_search.py

```diff
@@ -1050,16 +1051,31 @@ def resolve_field_list(fields, snuba_args, auto_fields=True):
     groupby that can be merged into the result of get_snuba_query_args()
     to build a more complete snuba query based on event search conventions.
     """
-    # If project.name is requested, get the project.id from Snuba so we
-    # can use this to look up the name in Sentry
-    if "project.name" in fields:
-        fields.remove("project.name")
-        if "project.id" not in fields:
-            fields.append("project.id")
-
     aggregations = []
     columns = []
     groupby = []
+
+    # If project is requested, we need to map ids to their names since snuba only has ids
+    if "project" in fields:
+        fields.remove("project")
+        if "project.id" not in fields:
+            fields.append("project.id")
+        project_ids = snuba_args["filter_keys"].get(
+            "project_id", []
+        )  # only gets used if sorting by project
+        projects = Project.objects.filter(id__in=project_ids).values("slug", "id")
+        aggregations.append(
+            [
+                "transform(project_id, [{}], {}, '')".format(
+                    # Need to use join like this so we don't get a list including Ls which confuses clickhouse
+                    ",".join([six.binary_type(project["id"]) for project in projects]),
+                    [six.binary_type(project["slug"]) for project in projects],
+                ),
+                None,
+                "project",
```

Comment:
TBH, not sure what the best column name here is. I like `project` since it matches the requested field. But I also see we use `project.name` in many places... 🤔

Reply:
It's probably easiest to leave it as project, since that will be the key in the response sent to the frontend, which is what they're expecting anyways.